notion-core/src/main/scala/com/ambiata/notion/core/SequenceFileIO.scala

```diff
@@ -0,0 +1,118 @@
+package com.ambiata.notion.core
+
+import com.ambiata.mundane.control._
+import com.ambiata.mundane.io._
+import com.ambiata.poacher.hdfs._
+import com.ambiata.saws.s3._
+import com.ambiata.com.amazonaws.services.s3.AmazonS3Client
+
+import org.apache.hadoop.io.compress.CompressionCodec
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.io.{Writable, BytesWritable, NullWritable, IntWritable, SequenceFile}
+import org.apache.hadoop.fs.{Hdfs => _, _}
+
+import java.io._
+
+import scalaz._, Scalaz._, effect.Effect._
+
+/**
+ * This is based off SequenceUtil, but can read/write sequence files of any key/value type
+ * 
+ * WARNING: Not suitable for reading/writing large amounts of data as everything is kept in memory
+ */
+case class SequenceFileIO[K <: Writable, V <: Writable, A](conf: Configuration, emptyKey: K, emptyValue: V, fromKeyValue: (K, V) => A, toKeyValue: A => (K, V))
```

Comment:
Should these live directly on the functions? You're only using `empty*` and `fromKeyValue` in read* and `toKeyValue` in write*